### PR TITLE
Implement Accelerator generating a skeleton of an Angular-based single page UI application

### PR DESCRIPTION
**Pull request by dmbelyaev**
Please also pay attention to the README. It describes the deployment topology.

### CODE REVIEW

File: angular-frontend/accelerator.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  options:
+    - name: artifactId
+      inputType: text
+      defaultValue: "client-side-ui-starter"
+      label: Application artifact name
+      required: true
+    - name: backendService
+      inputType: text
+      defaultValue: "customer-profile-backend.namespace.svc.cluster.local"
+      label: Hostname of the backend services app. If backend services are deployed to the same cluster then a local route can be used (e.g. <backend-app>.<namespace>.svc.cluster.local).
```
Comment (gm2552) [inline angular-frontend/accelerator.yaml:18]: Do you need the entire svc.cluster.local domain suffix for local cluster deployments?  Would it be easier to just use <backend-app>.<namespace>?

File: angular-frontend/accelerator.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  chain:
+    - merge:
+        - include: [ "**" ]
+          exclude: [ "nginx.conf" ]
+        - include: [ "nginx.conf" ]
+          chain:
+            - type: ReplaceText
+              substitutions:
+                - text: "http://backend/api/"
+                  with: "'http://' + #backendServiceResourceName + '/api/'"
```
Comment (gm2552) [inline angular-frontend/accelerator.yaml:36]: Is it safe to assume that all backing services are going to have a URI template that includes "/api/"?  Is this too brittle of a pattern?  Should the accelerator require the full service URL?

File: angular-frontend/accelerator.yaml (hunk 1 above)
Comment (dmbelyaev) [reply angular-frontend/accelerator.yaml:18]: I think we tried '<backend-app>.<namespace>' without the suffix and it didn't work. I'll check it again to be sure.

Comment (dmbelyaev) [reply angular-frontend/accelerator.yaml]: It works! Thanks for the hint!

File: angular-frontend/accelerator.yaml (hunk 2 above)
Comment (dmbelyaev) [reply angular-frontend/accelerator.yaml:36]: In the NGINX configuration forwarding is defined as:
```
location /api/ {
  proxy_pass http://backend/api/;
}
```

If '/api' is not included in the target location then NGINX will not add it. If services are deployed on another path then frontend developers will need to adapt the NGINX configuration.

**Review by gm2552 — APPROVED** [review]

File: angular-frontend/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+# angular-frontend
+
+angular-frontend provides you an out-of-the-box application setup to fast start development of a Web Application based
```
Comment (gm2552) [inline angular-frontend/README.md:3]: Would be good to add a narrative about the use case this accelerator is trying model:
   - A front end developer decoupled from the backend.

**Pull request merged**